File: src/features/billers/data/biller.ts
export type BillerMode = 'presentment' | 'openAmount';

export interface BillerCategory {
  readonly id: string;
  readonly name: string;
}

export interface BillerInputParam {
  readonly key: string;
  readonly label: string;
  readonly hint: string;
}

export interface Biller {
  readonly id: string;
  readonly categoryId: string;
  readonly name: string;
  readonly mode: BillerMode;
  readonly inputParams: readonly BillerInputParam[];
}

File: lib/features/billers/data/biller.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'biller.freezed.dart';

enum BillerMode { presentment, openAmount }

@freezed
abstract class BillerCategory with _$BillerCategory {
  const factory BillerCategory({
    required String id,
    required String name,
  }) = _BillerCategory;
}

@freezed
abstract class BillerInputParam with _$BillerInputParam {
  const factory BillerInputParam({
    required String key,
    required String label,
    required String hint,
  }) = _BillerInputParam;
}

@freezed
abstract class Biller with _$Biller {
  const factory Biller({
    required String id,
    required String categoryId,
    required String name,
    required BillerMode mode,
    required List<BillerInputParam> inputParams,
  }) = _Biller;
}
